### PR TITLE
Deployment manager failed due to BMC secret

### PR DESCRIPTION
**Pull request by MahalakshmiManthiram**
Serializing the IncompleteSecret to generate correct configuration. Updated the test cases.

Test plan:
- 'make' and 'make test' ran successfully
- 'deployctl tool generates the expected configuration

### CODE REVIEW

File: build/build.go [hunk 1]
```diff
@@ -95,9 +95,12 @@ func parseIncompleteSecret(secret *v1.Secret) *IncompleteSecret {
 	warningMsg := "Warning: Incomplete secret, please replace it with the secret content"
 	if secret.Type == v1.SecretTypeTLS {
```
Comment (wasnio) [inline build/build.go:96]: can you add to the yaml encode test case the SecretTypeBasicAuth and IncompleteSecret? Actually only the generic case is covered.

File: build/build_suite_test.go [hunk 2]
```diff
@@ -297,10 +296,7 @@ var _ = Describe("Test Build utilities:", func() {
 				}
 				got := parseIncompleteSecret(&secret)
 				expect := IncompleteSecret{
-					TypeMeta: metav1.TypeMeta{
```
Comment (wasnio) [inline build/build_suite_test.go:300]: I don't think we need to change here

File: build/build.go [hunk 3]
```diff
@@ -95,9 +95,12 @@ func parseIncompleteSecret(secret *v1.Secret) *IncompleteSecret {
...
 		return &IncompleteSecret{
-			TypeMeta:   secret.TypeMeta,
-			ObjectMeta: secret.ObjectMeta,
-			Type:       secret.Type,
+			TypeMeta: metav1.TypeMeta{},
```
[the remark this answers is not in the document]
Comment (MahalakshmiManthiram) [reply build/build.go:98]: This line was removed and tested.

File: build/build_suite_test.go (hunk 2 above)
Comment (MahalakshmiManthiram) [reply build/build_suite_test.go:300]: Removed the changes and tested.

**Review by wasnio — APPROVED** [review]

**Pull request merged**